azure/services/scalesetvms/scalesetvms.go

```diff
@@ -100,6 +127,89 @@ func (s *Service) Delete(ctx context.Context) error {
 	)
 	defer done()
 
+	if isFlex(instanceID) {
+		return s.deleteVMSSFlexVM(ctx, strings.TrimPrefix(providerID, azure.ProviderIDPrefix))
+	}
+	return s.deleteVMSSUniformInstance(ctx, resourceGroup, vmssName, instanceID, log)
+}
+
+func isFlex(instanceID string) bool {
+	return instanceID == ""
+}
+
+func (s *Service) deleteVMSSFlexVM(ctx context.Context, resourceID string) error {
+	ctx, log, done := tele.StartSpanWithLogger(ctx, "scalesetvms.Service.deleteVMSSFlexVM")
+	defer done()
+
+	defer func() {
+		if vm, err := s.VMClient.GetByID(ctx, resourceID); err == nil && vm.VirtualMachineProperties != nil {
+			log.V(4).Info("updating vmss vm state", "state", vm.ProvisioningState)
```

Comment:
Do we want to be more explicit here and suggest "delete in progress" rather than just the generic "updating vmss vm state"?